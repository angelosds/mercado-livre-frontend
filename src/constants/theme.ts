export const ROOT_RELATIVE_SIZE = 16
