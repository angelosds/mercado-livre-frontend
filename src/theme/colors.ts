import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  primary: '#fe6000',
  secondary: '#3483fa',
  gray900: '#333333',
  gray700: '#666666',
  gray500: '#999999',
  gray300: '#eeeeee',
  white: '#ffffff',
};
