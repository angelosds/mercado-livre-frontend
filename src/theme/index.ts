import { DefaultTheme } from 'styled-components';

import { pixelToRem } from '../utils';

export const theme: DefaultTheme = {
  colors: {
    primary500: '#fee600',

    secondary100: '#c6dcf7',
    secondary500: '#3483fa',
    secondary700: '#2968c8',

    gray900: '#333333',
    gray700: '#333333',
    gray500: '#666666',
    gray400: '#999999',
    gray300: '#bbbbbb',
    gray200: '#dddddd',
    gray100: '#eeeeee',

    border200: '#f7f7f7',

    success: '#1ed760',

    white: '#ffffff',
  },
  fontSizes: {
    xs: pixelToRem(14),
    sm: pixelToRem(16),
    md: pixelToRem(24),
    lg: pixelToRem(28),
    xl: pixelToRem(46),
  },
  outline: {
    sm: '3px'
  },
  space: {
    sm: '4px',
    md: '8px',
    lg: '16px',
    xl: '32px',
  },
  radius: {
    sm: '4px'
  },
  breakpoints: {
    mobile: '565px'
  }
};
