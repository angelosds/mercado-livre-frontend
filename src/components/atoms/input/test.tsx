import { fireEvent, render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect'

import { theme } from '../../../theme';

import Input, { Props } from '.'

describe('Input', () => {
  const build = ({ placeholder, value, onChange }: Props) => {
    const { container } = render(<ThemeProvider theme={theme}><Input placeholder={placeholder} value={value} onChange={onChange} /></ThemeProvider>)

    return { container }
  }

  it('should render the component', () => {
    const { container } = build({ value: '', onChange: () => {} })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a placeholder text', () => {
    const placeholder = 'hello world'

    build({ placeholder, onChange: () => {} })

    expect(screen.getByRole('textbox')).toHaveAttribute('placeholder', placeholder)
  })

  it('should add a value to the input', () => {
    const value = 'hello world'

    build({ value, onChange: () => {} })

    expect(screen.getByRole('textbox')).toHaveValue(value)
  })

  it('should emit onChange when input value changes', () => {
    const onChange = jest.fn()

    build({ onChange })

    fireEvent.change(screen.getByRole('textbox'), { target: { value: 'hello world' } })

    expect(onChange).toHaveBeenCalled()
  })
})
