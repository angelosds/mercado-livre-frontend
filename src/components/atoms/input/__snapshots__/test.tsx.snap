// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render the component 1`] = `
<input
  class="sc-bdvvtL gufCAc"
  value=""
/>
`;
